packages/polymath-ui/src/components/inputs/DatePickerInput.js

```diff
@@ -19,34 +20,37 @@ type Props = {
 };
 
 export default ({
-  input,
+  input: { name, onChange, onBlur },
   label,
   meta: { touched, error },
   className,
+  placeholder,
   ...rest
 }: Props) => {
   const invalid = touched && !!error;
   return (
     <DatePicker
-      id={input.name}
+      id={name}
       className={className}
       datePickerType="single"
       // eslint-disable-next-line
       onChange={(date: Date) => {
-        input.onChange(date || null);
+        onChange(date || null);
+        // redux-form updates `touched` on blur.
+        onBlur();
```

Comment:
I'm not convinced this is the best way to do this, it is missleading to call `onBlur()` if we are not losing focus to the component.

I'll play with this component this week to have some more context and offer a better solution if there is one.

Reply:
I have tried a few other options but seems this is a bug in redux-form. If you find a better way kindly let me know

Reply:
Will do @sajclarke I'll provide a solution if there is one as soon as I can.

Reply:
I had a look at this and I think it's the only way. I think it's fine to do this because it doesn't actually blur the field. It just calls "onBlur" to Redux-form as that's the only way to trigger the validation.